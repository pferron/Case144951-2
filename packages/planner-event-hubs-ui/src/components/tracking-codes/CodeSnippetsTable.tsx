import React, { Dispatch, SetStateAction } from 'react';
import { Table, TableColumn } from '@cvent/carina/components/Table';
import { AccountCodeSnippet } from '@cvent/planner-event-hubs-model/types';
import { useTranslate } from 'nucleus-text';

interface Props {
  dataList: AccountCodeSnippet[];
  selected: string;
  setSelected: Dispatch<SetStateAction<string>>;
}
function CodeSnippetsTable({ dataList, selected, setSelected }: Props): JSX.Element {
  const { translate } = useTranslate();

  const handleOnSelect = (rowName, isSelected): void => {
    if (isSelected) setSelected(null);
    else setSelected(rowName);
  };

  const snippetsArray = dataList.map(codeSnippet => ({
    ...codeSnippet,
    rowName: `${codeSnippet.codeSnippetId}`,
    visitorsCanTurnOff: codeSnippet.isDropCodeSnippetToCookieBannerTied
      ? translate('code_snippets_table_disable_code_snippets_option_yes')
      : translate('code_snippets_table_disable_code_snippets_option_no')
  }));

  return (
    <Table
      data={snippetsArray}
      rowMode="multi-select"
      testID="code-snippet-table"
      selected={[selected]}
      onSelect={(rowName, { isSelected }) => {
        handleOnSelect(rowName, isSelected);
      }}
      onMultiSelectAction={_val => {
        handleOnSelect(_val, true);
      }}
    >
      <TableColumn name="codeSnippetName" heading={translate('code_snippets_table_header_name')} />
      <TableColumn name="codeSnippetDataTagCode" heading={translate('code_snippets_table_header_datatag_code')} />
      <TableColumn name="codeSnippetStatus" heading={translate('code_snippets_table_header_status')} />
      <TableColumn name="visitorsCanTurnOff" heading={translate('code_snippets_table_header_disable_code_snippets')} />
    </Table>
  );
}

export default CodeSnippetsTable;
