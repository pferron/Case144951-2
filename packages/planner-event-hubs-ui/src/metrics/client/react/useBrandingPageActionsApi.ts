/**
 * ------------------------------------------------------------------------------------
 * This client was automatically generated. ** Do Not Edit **
 * ------------------------------------------------------------------------------------
 */
import { useCallback } from 'react';
import * as Rudder from '../rudder-web';
import { usePageActions } from '@cvent/sli-nextjs-metrics';
import { ApiCallback } from '@rudderstack/analytics-js';
import { useIdentityApi } from '../IdentityProvider';
import { TrackProperties } from '../commonTypes';

export type ActionDefinition = {
  id: string;
  destinations?: string[];
};

export interface BrandingSaveButtonClicked extends TrackProperties {
  // Primary color of the events+ hub
  primaryColor: string;
  // Secondary color color of the events+ hub
  secondaryColor: string;
  // BackgroundColor color color of the events+ hub
  backgroundColor: string;
  // Mood of the events+ hub
  selectYourMood: string;
  // Whether safemode is turned on or off for the events+ hub
  safeColorMode: boolean;
  // Whether background color is overriden for the events+ hub
  overrideBackgroundColor?: boolean;
}
export interface LogoImageChecked extends TrackProperties {}

interface AnalyticsApi {
  brandingSaveButtonClicked: (
    props: BrandingSaveButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  logoImageChecked: (props?: LogoImageChecked, options?: Rudder.Options, callback?: ApiCallback) => void;
}

/**
 * This hook exposes a typed api with functions from the application's tracking plan.
 * Internally, each function uses the usePageActions API to track actions and forward
 * them to the reporters that were installed during the Metrics initialization in your
 * application's main entry point component.
 *
 * Note: This is the recommended way of tracking/measuring actions as it allows you to
 * register base properties during the metrics' client initialization, which will be included
 * automatically to every action whenever they are executed by your instrumentation code
 *
 *
 * How to use it:
 *
 * `const { acqSignupCompleted } = useBrandingPageActionsApi();`
 *
 *  // Somewhere in your application you can use this function to track this action
 * `acqSignupCompleted({ userId, email, accountName });`
 *
 * // If you want to track this action as an error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, error: '500 Error........' });`
 *
 * // If you want to track this action as an expected error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, expected_error: 'This email address is already associated with an existing account.' });`
 *
 * @returns AnalyticsApi
 */
export const useBrandingPageActionsApi = (): AnalyticsApi => {
  const { trackAction, getBaseProps } = usePageActions();
  const { trackingPlanName } = getBaseProps() || {};
  const { updateProfile } = useIdentityApi();

  /**
   * this action tracks the users who makes changes in themes in order to change the colors of branding
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : branding save button clicked
   * downstreamName: branding save button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when clicks on save button in brandind edit page
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BrandingSaveButtonClicked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const brandingSaveButtonClicked = useCallback(
    (props: BrandingSaveButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'branding save button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'branding save button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who saves the image in order to use as logo in events+
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : logo image checked
   * downstreamName: logo image checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of website image section in branding section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {LogoImageChecked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const logoImageChecked = useCallback(
    (props?: LogoImageChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'logo image checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'logo image checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );

  return {
    brandingSaveButtonClicked,
    logoImageChecked
  };
};
