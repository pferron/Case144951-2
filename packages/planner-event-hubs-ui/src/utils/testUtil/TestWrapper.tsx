import React, { useMemo } from 'react';
import { LocalizationProvider } from 'nucleus-text';
import { Breakpoints } from '@cvent/carina/components/Breakpoints';
import { AppFeature } from '@cvent/planner-event-hubs-model/types';
import { ThemeProvider } from '@cvent/carina/components/ThemeProvider/ThemeProvider';
import { getDefaultTheme } from '@cvent/carina/utils/tokens';
import { AppFeaturesProvider } from '@components/AppFeaturesProvider';
import { UserPermissionsContext } from '@hooks/UserPermissionsProvider';
import locales, { defaultLocale } from '../../../locales';

export function TestWrapper({ children, appFeatures }: Props): JSX.Element {
  const userPermissionsValue = useMemo(() => ({ VideoCenter: true }), []);
  return (
    <ThemeProvider theme={getDefaultTheme()}>
      <LocalizationProvider locale={defaultLocale} locales={locales}>
        <Breakpoints ssrSize="m" key={typeof window === 'undefined' ? 'ssr' : 'csr'}>
          <AppFeaturesProvider value={appFeatures}>
            <UserPermissionsContext.Provider value={userPermissionsValue}>{children}</UserPermissionsContext.Provider>
          </AppFeaturesProvider>
        </Breakpoints>
      </LocalizationProvider>
    </ThemeProvider>
  );
}

interface Props {
  children: JSX.Element;
  appFeatures?: AppFeature[];
}
