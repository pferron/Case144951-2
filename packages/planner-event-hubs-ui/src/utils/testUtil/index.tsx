import { render, configure } from '@testing-library/react';
import { TestWrapper } from './TestWrapper';

configure({ testIdAttribute: 'data-cvent-id' });

type RenderParams = Parameters<typeof render>;
type RenderReturn = ReturnType<typeof render>;
const customRender = (ui: RenderParams[0], options?: RenderParams[1]): RenderReturn =>
  render(ui, { wrapper: TestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
