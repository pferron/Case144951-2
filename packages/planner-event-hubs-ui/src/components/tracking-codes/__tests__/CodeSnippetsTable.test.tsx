import React from 'react';
import { TestWrapper } from '@utils/testUtil/TestWrapper';
import { render, fireEvent, screen } from '@testing-library/react';
import CodeSnippetsTable from '@components/tracking-codes/CodeSnippetsTable';
import { accountCodeSnippet } from '@components/tracking-codes/fixtures/CodeSnippetData';
import 'jest-axe/extend-expect';

describe('CodeSnippetsTable', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const mockFn = jest.fn();

  it('Renders CodeSnippets Table with headers', async () => {
    const { baseElement } = render(
      <TestWrapper>
        <CodeSnippetsTable dataList={[]} selected={null} setSelected={mockFn} />
      </TestWrapper>
    );

    expect(baseElement).toMatchSnapshot();
    expect(await screen.findByText('Name')).toBeInTheDocument();
    expect(await screen.findByText('Data Tag Code')).toBeInTheDocument();
    expect(await screen.findByText('Status')).toBeInTheDocument();
    expect(await screen.findByText('Visitors can turn off')).toBeInTheDocument();
    // HUB-151797
    // expect(await axe(baseElement)).toHaveNoViolations();
  });

  it('Renders CodeSnippets Table with data', async () => {
    const { baseElement } = render(
      <TestWrapper>
        <CodeSnippetsTable dataList={accountCodeSnippet} selected={null} setSelected={mockFn} />
      </TestWrapper>
    );

    expect(baseElement).toMatchSnapshot();
    expect(await screen.findByText('TEST')).toBeInTheDocument();
    expect(await screen.findByText('Test-code-snippet')).toBeInTheDocument();
    expect(await screen.findByText('TEST2')).toBeInTheDocument();
    expect(await screen.findByText('Marketo2')).toBeInTheDocument();
    // HUB-151797
    // expect(await axe(baseElement)).toHaveNoViolations();
  });

  it('CodeSnippets on select', async () => {
    render(
      <TestWrapper>
        <CodeSnippetsTable dataList={accountCodeSnippet} selected={null} setSelected={mockFn} />
      </TestWrapper>
    );

    const element = await screen.findByText('Test-code-snippet');
    fireEvent.click(element);
    expect(mockFn).toHaveBeenCalledWith('Test-code-snippet-id');
    // HUB-151797
    // expect(await axe(baseElement)).toHaveNoViolations();
  });
});
