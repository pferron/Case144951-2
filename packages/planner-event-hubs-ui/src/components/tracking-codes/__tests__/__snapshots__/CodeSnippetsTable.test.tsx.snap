// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeSnippetsTable Renders CodeSnippets Table with data 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  padding: 1px;
}

.emotion-1 {
  width: 100%;
  box-sizing: border-box;
}

.emotion-2 {
  box-sizing: border-box;
  outline-width: 1px;
  outline-offset: -1px;
  outline-style: solid;
  outline-color: rgb(217,220,222);
  width: 100%;
  overflow: hidden;
  min-height: 56px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 8px;
  padding-right: 8px;
  background-color: rgb(255,255,255);
}

.emotion-3 {
  box-sizing: border-box;
  position: relative;
  width: 54px;
  background-color: rgb(255,255,255);
  padding-left: 8px;
  padding-right: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  font-size: 0.875rem;
  font-family: 'Rubik','Helvetica','Arial';
  min-width: 54px;
}

.emotion-4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  outline: none;
  border: none;
  box-shadow: none;
  background-color: transparent;
  padding: 0;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.emotion-5 {
  box-sizing: border-box;
  position: relative;
  width: 0;
  background-color: rgb(255,255,255);
  padding-left: 8px;
  padding-right: 8px;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 0.875rem;
  font-family: 'Rubik','Helvetica','Arial';
  min-width: 150px;
}

.emotion-6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  outline: none;
  border: none;
  box-shadow: none;
  background-color: transparent;
  padding: 0;
  font-size: 0.875rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgb(26,32,38);
}

.emotion-6:focus>.heading-text::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  top: calc(100% + 1px);
  bottom: -2px;
  background-color: rgb(0,106,225);
  pointer-events: none;
}

.emotion-7 {
  position: relative;
  font-weight: 400;
  font-family: 'Rubik','Helvetica','Arial';
  overflow: hidden;
}

.emotion-17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background-color: rgb(217,220,222);
  border-top-width: 1px;
  border-top-style: solid;
  border-top-color: rgb(217,220,222);
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: rgb(217,220,222);
}

.emotion-18 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: rgb(105,113,122);
  width: 100%;
  overflow-x: hidden;
  -ms-overflow-y: hidden;
  border-left: 1px solid rgb(217,220,222);
  border-right: 1px solid rgb(217,220,222);
}

.emotion-18::before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 3;
  pointer-events: none;
}

.emotion-18:hover button[aria-checked],
.emotion-18:focus-within button[aria-checked] {
  z-index: 5;
}

.emotion-18:hover button[aria-checked]>div,
.emotion-18:focus-within button[aria-checked]>div {
  border-color: rgba(154,159,166,0.8);
  background-color: transparent;
}

.emotion-18:hover button[aria-checked]>div svg,
.emotion-18:focus-within button[aria-checked]>div svg {
  fill: rgba(154,159,166,0.8);
}

.emotion-18:hover span:first-child:hover button[aria-checked]>div,
.emotion-18:focus-within span:first-child:hover button[aria-checked]>div,
.emotion-18:hover span:first-child:focus-within button[aria-checked]>div,
.emotion-18:focus-within span:first-child:focus-within button[aria-checked]>div {
  border-color: rgb(255,255,255);
}

.emotion-18:hover span:first-child:hover button[aria-checked]>div svg,
.emotion-18:focus-within span:first-child:hover button[aria-checked]>div svg,
.emotion-18:hover span:first-child:focus-within button[aria-checked]>div svg,
.emotion-18:focus-within span:first-child:focus-within button[aria-checked]>div svg {
  fill: rgb(255,255,255);
}

.emotion-18 .x-ray {
  z-index: 5;
  pointer-events: none;
  opacity: 0;
  -webkit-transition: opacity .18s;
  transition: opacity .18s;
  position: absolute;
  -webkit-transform: translateY(2px);
  -moz-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
  -ms-transform: translate(-22px, -8px);
}

@media (hover: hover) and (pointer: fine) {
  .emotion-18:hover .x-ray.start,
  .emotion-18:hover .x-ray.end {
    opacity: 1;
  }
}

@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  .emotion-18:hover .x-ray.start,
  .emotion-18:hover .x-ray.end {
    opacity: 1;
  }
}

.emotion-19 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(255,255,255);
  width: 100%;
  padding-left: 8px;
  padding-right: 8px;
  z-index: 1;
  min-height: 56px;
  padding-top: 10px;
  padding-bottom: 10px;
  left: 0;
  -webkit-transition: left .18s;
  transition: left .18s;
  right: unset;
  outline-width: 1px;
  outline-style: solid;
  outline-color: rgb(217,220,222);
  border-top: 1px solid rgb(217,220,222);
}

.emotion-19:focus {
  outline: none;
}

.emotion-19::after {
  content: "";
  position: absolute;
  top: 0;
  bottom: 1px;
  left: 1px;
  right: 1px;
  pointer-events: none;
}

@media (hover: hover) and (pointer: fine) {
  .emotion-19:hover {
    background-color: #E7F2FF;
  }
}

.emotion-20 {
  box-sizing: border-box;
  width: 54px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: rgb(26,32,38);
  font-family: 'Rubik','Helvetica','Arial';
  font-weight: 300;
  font-size: 0.875rem;
  padding-right: 8px;
  padding-left: 8px;
  min-width: 54px;
}

.emotion-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-22 {
  box-sizing: border-box;
  height: 16px;
  width: 16px;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(105,113,122);
  border-radius: 4px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.emotion-23 {
  box-sizing: border-box;
  width: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  color: rgb(26,32,38);
  font-family: 'Rubik','Helvetica','Arial';
  font-weight: 300;
  font-size: 0.875rem;
  padding-right: 8px;
  padding-left: 8px;
  min-width: 150px;
}

.emotion-24 {
  max-width: 100%;
  overflow: visible;
  overflow-wrap: anywhere;
}

<body>
  <div>
    <div
      dir="ltr"
    >
      <div
        class="emotion-0"
        data-cvent-id="code-snippet-table"
        dir="ltr"
        role="table"
      >
        <div
          class="emotion-1"
          data-cvent-id="code-snippet-table-table-header"
          role="rowgroup"
        >
          <div
            class="emotion-2"
            role="row"
          >
            <div
              class="emotion-3"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-4"
              />
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-1"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Name
                </div>
              </div>
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-2"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Data Tag Code
                </div>
              </div>
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-3"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Status
                </div>
              </div>
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-4"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Visitors can turn off
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-17"
          role="rowgroup"
        >
          <div
            class="emotion-18"
          >
            <div
              class="emotion-19"
              data-cvent-id="code-snippet-table-row-0"
              role="row"
              tabindex="0"
            >
              <div
                class="emotion-20"
                role="cell"
              >
                <div
                  class="emotion-21"
                >
                  <div
                    class="emotion-22"
                  />
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  Test-code-snippet
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  TEST
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  Approved
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  Yes
                </div>
              </div>
            </div>
          </div>
          <div
            class="emotion-18"
          >
            <div
              class="emotion-19"
              data-cvent-id="code-snippet-table-row-1"
              role="row"
              tabindex="0"
            >
              <div
                class="emotion-20"
                role="cell"
              >
                <div
                  class="emotion-21"
                >
                  <div
                    class="emotion-22"
                  />
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  Marketo2
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  TEST2
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  Approved
                </div>
              </div>
              <div
                class="emotion-23"
                role="cell"
              >
                <div
                  class="emotion-24"
                >
                  Yes
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`CodeSnippetsTable Renders CodeSnippets Table with headers 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  padding: 1px;
}

.emotion-1 {
  width: 100%;
  box-sizing: border-box;
}

.emotion-2 {
  box-sizing: border-box;
  outline-width: 1px;
  outline-offset: -1px;
  outline-style: solid;
  outline-color: rgb(217,220,222);
  width: 100%;
  overflow: hidden;
  min-height: 56px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 8px;
  padding-right: 8px;
  background-color: rgb(255,255,255);
}

.emotion-3 {
  box-sizing: border-box;
  position: relative;
  width: 54px;
  background-color: rgb(255,255,255);
  padding-left: 8px;
  padding-right: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  font-size: 0.875rem;
  font-family: 'Rubik','Helvetica','Arial';
  min-width: 54px;
}

.emotion-4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  outline: none;
  border: none;
  box-shadow: none;
  background-color: transparent;
  padding: 0;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.emotion-5 {
  box-sizing: border-box;
  position: relative;
  width: 0;
  background-color: rgb(255,255,255);
  padding-left: 8px;
  padding-right: 8px;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 0.875rem;
  font-family: 'Rubik','Helvetica','Arial';
  min-width: 150px;
}

.emotion-6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  outline: none;
  border: none;
  box-shadow: none;
  background-color: transparent;
  padding: 0;
  font-size: 0.875rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgb(26,32,38);
}

.emotion-6:focus>.heading-text::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  top: calc(100% + 1px);
  bottom: -2px;
  background-color: rgb(0,106,225);
  pointer-events: none;
}

.emotion-7 {
  position: relative;
  font-weight: 400;
  font-family: 'Rubik','Helvetica','Arial';
  overflow: hidden;
}

.emotion-17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background-color: rgb(217,220,222);
  border-top-width: 1px;
  border-top-style: solid;
  border-top-color: rgb(217,220,222);
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: rgb(217,220,222);
}

<body>
  <div>
    <div
      dir="ltr"
    >
      <div
        class="emotion-0"
        data-cvent-id="code-snippet-table"
        dir="ltr"
        role="table"
      >
        <div
          class="emotion-1"
          data-cvent-id="code-snippet-table-table-header"
          role="rowgroup"
        >
          <div
            class="emotion-2"
            role="row"
          >
            <div
              class="emotion-3"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-4"
              />
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-1"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Name
                </div>
              </div>
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-2"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Data Tag Code
                </div>
              </div>
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-3"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Status
                </div>
              </div>
            </div>
            <div
              class="emotion-5"
              role="columnheader"
              scope="col"
            >
              <div
                class="emotion-6"
                data-cvent-id="code-snippet-table-heading-4"
              >
                <div
                  class="heading-text emotion-7"
                >
                  Visitors can turn off
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-17"
          role="rowgroup"
        />
      </div>
    </div>
  </div>
</body>
`;
